test/index-spec.js

```diff
@@ -0,0 +1,55 @@
+import * as mocha from 'mocha';
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import io from 'socket.io-client';
+
+import Server from '../src/server';
+
+const { assert, expect } = chai;
+const options = null;
+const user1 = { name: 'user1' };
+
+const server = new Server();
+server.start()
+  .then(() => {
+    chai.use(chaiHttp);
+
+    mocha.describe('GET /echo test', () => {
+      mocha.it('should echo hello', (done) => {
+        const echoUrl = '/echo?msg=hello';
+        chai.request(server.app)
+          .get(echoUrl)
+          .end((err, res) => {
+            expect(res).to.have.status(200);
+            assert.equal(res.text, 'hello');
+            done();
+          });
+      });
+    });
+
+    mocha.describe('Event Message Test', () => {
+      // "dial" : A가 시그널링 서버한테 통화 요청
+      mocha.describe('Dial', () => {
+        mocha.it('user1 should dial to Signaling server', (done) => {
+          const url = `http://localhost:${server.port}`;
+          const client1 = io.connect(url, options);
+
```

Comment:
I love this code but we need to solve callback hell problem.
What do you think about it?

Reply:
For solving callback hell problem, There are two concept in my head at this point
1. Go lang.
This article says "callbacks suck"
(https://sangwook.github.io/2014/07/06/farewell-node-js-tj-holowaychuk.html)
2. Promise
I saw this article
(https://victorydntmd.tistory.com/49)

As we are using node.js, We can't avoid using callbacks.
and I don't know ES6 well, these are all I know